datafusion/core/src/test/mod.rs

```diff
@@ -38,6 +38,7 @@ use crate::test_util::{aggr_test_schema, arrow_test_data};
 use arrow::array::{self, Array, ArrayRef, Decimal128Builder, Int32Array};
 use arrow::datatypes::{DataType, Field, Schema};
 use arrow::record_batch::RecordBatch;
+#[cfg(feature = "compression")]
 use datafusion_common::DataFusionError;
```

Comment:
This is only used by compression. Got warning via 
```
cargo check --profile ci --no-default-features -p datafusion --features=parquet_encryption
```
Figured I may as well fix it while I am here.